enhancements/kube-apiserver/stability-api-removal-notifications.md

```diff
@@ -0,0 +1,162 @@
+---
+title: api-removal-notifications
+authors:
+  - "@sanchezl"
+reviewers:
+  - "@deads2k"
+  - TBD
+approvers:
+  - TBD
+creation-date: 2021-02-01
+last-updated: 2021-02-01
+status: provisional
+---
+
+# API Removal Notifications
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Notify customers that an API that will be removed in the next release is in use.
+
+## Motivation
+
+Kubernetes is being more aggressive with the removal of APIs that have been deprecated or pre-release APIs that have
+reached a dead-end and will never reach GA. Additionally, the ability to re-enable a previously removed API has been
+removed.
+
+Customers should be notified that an API that will be removed in the next release is in use so that they can decide if
+there is a need to take action in order to ensure success when upgrading to the next release.
+
+### Goals
+
+1. Notify customers that an API that will be removed in the next release is in use.
```

Comment:
Is there a reason this cannot be expanded to the full three minor versions required of the deprecation policy? Are we concerned that there's too much capitulation here, as in we'd say something is removed in 1.24 today but there's risk that it is not? To me that's tolerable, we'd just update the ruleset when we become aware of that change and it's only an Info alert.

Reply:
While building a mechanism to see those is reasonable, we think that building a mechanism to alert on them, even at an info level, would teach customers that we alert on things that won't be a problem for a year+ and teach them to ignore these warnings.

Given the API resources being added in this design, there is a mechanism for an interested party to look at usage of APIs slated for future removal.